### PR TITLE
Add search in xata client

### PR DESCRIPTION
**Pull request by SferaDev**
Closes: #164

### CODE REVIEW

File: client/src/schema/repository.ts [hunk 1]
```diff
@@ -517,12 +539,25 @@ export class BaseClient<D extends Record<string, Repository<any>> = Record<strin
...
       }
     });
   }
 
+  async search<Tables extends keyof D>(
+    query: string,
+    tables: Tables[] = Object.keys(this.db) as Tables[],
+    options?: { fuzziness?: number }
+  ): Promise<{ [Model in GetArrayInnerType<typeof tables>]: Awaited<ReturnType<D[Model]['search']>> }> {
+    // TODO: Implement global search with a single call, REST repository abstraction needed
```
Comment (SferaDev) [inline client/src/schema/repository.ts:553]: We need to revisit this, but a discussion on plugins/namespaces should happen before. We can discuss it in our next 1:1 or a team sync.

File: test/integration.test.ts [hunk 2]
```diff
@@ -817,3 +824,44 @@ describe('getBranch', () => {
...
+
+    expect(users[0].id).toBeDefined();
+    expect(users[0].read).toBeDefined();
+    expect(users[0].full_name?.includes('fruits')).toBeTruthy();
+
+    expect(teams[0].id).toBeDefined();
+    expect(teams[0].read).toBeDefined();
+    expect(teams[0].name?.includes('fruits')).toBeTruthy();
+  });
+});
```
Comment (SferaDev) [inline test/integration.test.ts:867]: This randomly fails (indexes not created yet, API returns empty array)

Comment (SferaDev) [inline test/integration.test.ts:867]:
```suggestion
```

Comment (SferaDev) [reply test/integration.test.ts:867]: Added them back, seems there was a bit of delay yesterday.

Creating https://github.com/xataio/client-ts/issues/181 for follow-up

File: client/src/schema/repository.ts (hunk 1 above)
Comment (SferaDev) [reply client/src/schema/repository.ts:557]: I would like to offer both (inside and outside) as-is here.

Creating https://github.com/xataio/client-ts/issues/182 for improving this

**Comment by SferaDev** [comment]
@gimenete Moving this again for review, I will follow-up on the improvements soon.

What do you think about the API in any case?

```ts
const users = await xata.db.users.search("Alberto")
const { users } = await xata.search("Alberto", { tables: "users" })
```

**Comment by gimenete** [comment]
> What do you think about the API in any case?

I think that looks great 👍

**Review by gimenete — APPROVED** [review]
LGTM 👌

**Pull request merged**